### ISSUE TITLE
Certain compile-time errors cause 'No exception handler located for catch' error

### ISSUE THREAD

**Issue by vrurg** [rakudo/rakudo#2739]
## The Problem

Certain exceptions raised at compile time causes rakudo to bail out with _'No exception handler located for catch'_ error. 

## Expected Behavior

The initial exception must be printed to the console.

## Steps to Reproduce

A code which causes this kind behavior to me follows:

```
sub foo ( ::T $val ) {
    my T $a is default($val);
}
foo(42);
```

## Notes

The source of the problem is currently narrowed down to `Exception::print_error()`. The fix is being worked on.

## Environment
* Operating system: macOS 10.14
* Compiler version (`perl6 -v`): _This is Rakudo version 2018.12 built on MoarVM version 2018.12_, _master_ is affected too.

**Comment by JJ** [comment]
I have the same problem in Test::Output. Referenced #2751

**Comment by vrurg** [comment]
@JJ It is similar yet different problem. The only commonality between the two is the fact that rakudo leaks an exception and then complains about this. My fix does close one case when such leakage happens. More generic fix should report the leaked exception alongside with 'No handler' message to improve diagnostics.

**Comment by vrurg** [comment]
Closed via #2740
